game-core/src/main/java/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1171,8 +1168,8 @@ private static MoveValidationResult validateTransport(
     }
     // If there are non-sea transports return
     final boolean seaOrNoTransportsPresent =
```

Comment:
The variable name and the comment doesn't match the behaviour🤔 
It should probably be called `seaTransportPresent`.
In any case the expression can be simplified:
- `!seaOrNoTransportsPresent`
- == `!(unitsToTransports.isEmpty() || unitsToTransports.values().stream().anyMatch(Matches.unitIsSea().and(Matches.unitCanTransport())))`
- == `!unitsToTransports.isEmpty() && unitsToTransports.values().stream().noneMatch(Matches.unitIsSea().and(Matches.unitCanTransport()))`